File: python/anagram/anagram.py
class Anagram(object):
  
    def __init__(self, word):
        self.word = word

    def match(self, candidates):
        formatted_word = FormattedWord(self.word)
        return [candidate for candidate in candidates 
            if FormattedWord(candidate).is_anagram_of(formatted_word)]


class FormattedWord(object):
  
    def __init__(self, word):
        self._lower = word.lower();
        self._sorted = "".join(sorted(self._lower))

    def is_anagram_of(self, other_word):
        return self._lower!= other_word._lower and self._sorted == other_word._sorted

File: javascript/anagram/anagram.js
function Anagram(word) {

  this.word = new FormattedWord(word);

  this.match = function(candidates) {
    var anagrams = [];
    for (var i = 0; i < candidates.length; i++) {
      var currentWord = candidates[i];
      var currentCandidate = new FormattedWord(currentWord);
      if (this.word.isAnagramOf(currentCandidate)) {
        anagrams.push(currentWord);
      };
    };
    return anagrams;
  };

}

function FormattedWord(word) {

  this.standardized = word.toLowerCase();
  this.sorted = this.standardized.split("").sort().join("");

  this.isAnagramOf = function(otherWord) {
    return this.standardized != otherWord.standardized && this.sorted == otherWord.sorted;
  };

}

module.exports = Anagram;